packages/gated-content/src/conditions/index.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  AccessCondition,
+  AnyCriterionFragment,
+  LeafCriterionFragment,
+} from '@lens-protocol/api-bindings';
+import { InvariantError, isNonNullable, never, UnknownObject } from '@lens-protocol/shared-kernel';
+import { UnifiedAccessControlConditions } from '@lit-protocol/constants';
+
+import { EnvironmentConfig } from '../environments';
+import { Entry } from '../types';
+import { transformCollectCondition } from './collect-condition';
+import { transformEoaCondition } from './eoa-condition';
+import { transformErc20Condition } from './erc20-condition';
+import { transformFollowCondition } from './follow-condition';
+import { transformNftCondition } from './nft-condition';
+import { transformProfileCondition } from './profile-condition';
+import {
+  AccessConditionType,
+  isAccessConditionType,
+  LitAccessControlCondition,
+  LitNestedAccessControlCondition,
+} from './types';
+import { insertObjectInBetweenArrayElements } from './utils';
+import {
+  assertAtLeastTwoCriteria,
+  assertNoMoreThanFiveCriteria,
+  InvalidAccessCriteriaError,
+} from './validators';
+
+function extractConditionEntry<T extends UnknownObject>(condition: T): Entry<T> {
+  for (const key in condition) {
+    if (isAccessConditionType(key) && isNonNullable(condition[key as keyof T])) {
+      return [key, condition[key] ?? never()];
+    }
+  }
+  throw new InvariantError('Invalid access condition');
+}
+
+function flatten<T>(conditions: T[]): T[] {
```

Comment:
can we just use array `flatten`?

Reply:
This one flattens it only if the nested array has `lenght==1`, not exactly the same as lodash-style 'flatten` function.